### PR TITLE
Don't try to call readlink() on regular files as part of files_select

### PR DESCRIPTION
**Pull request by estenberg**
Redmine #6996 (also has acceptance test).

### CODE REVIEW

File: cf-agent/files_select.c [hunk 1]
```diff
@@ -494,6 +494,11 @@ static int SelectIsSymLinkTo(EvalContext *ctx, char *filename, Rlist *crit)
             Log(LOG_LEVEL_DEBUG, "Skipping readlink() on directory %s", filename);
             return false;
         }
+        else if (S_ISREG(statbuf.st_mode))
```
Comment (kacfengine) [inline cf-agent/files_select.c]: I would rather change consolidate both of these into this:

```
else if (!S_ISLNK(statbuf.st_mode))
{
    Log(LOG_LEVEL_DEBUG, "Skipping readlink() on non-symlink %s", filename);
    return false;
}
```

Then you catch other error cases such as device files, sockets and fifos. We've been getting several fixes lately for these cases by @worr, for example.

**Comment by kacfengine** [comment]
You're missing a Changelog entry for this. I strongly urge you to run ``core/misc/githooks/install-hooks`` since it will install Git hooks that will remind you.

When amending this commit you'll have to add it manually though, since the commit message is preexisting. It takes this form:

```
Don't try to call readlink() on regular files as part of files_select
Redmine #6996.

Changelog: <message>
```

Remember that changelog messages are user facing, so you should describe how the behavior has changed, not what you did to fix it.

**Comment by estenberg** [comment]
OK, thanks, updated.

**Comment by estenberg** [comment]
Acceptance test: #2114

**Comment by kacfengine** [comment]
Ah, does this fix the acceptance test? Then you should remove the ``"test_soft_fail"`` portion of that test in the same commit, because it inverts the pass/fail relationship, so if it starts passing, the result is fail. :-) This is so that we learn of tests that start passing and mark them appropriately.

**Comment by estenberg** [comment]
Alright, trying again.

**Comment by kacfengine** [comment]
Thanks!

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
Thanks, merged manually.

**Comment by kacfengine** [comment]
Cherry-picked to 3.6.x and 3.7.x as well.

**Pull request closed without merging**